src/main/java/org/akaza/openclinica/control/core/SecureController.java

```diff
@@ -670,6 +665,15 @@ else if (currentRole.getId() > 0
             mayProceed();
             // pingJobServer(request);
             // Set if enrollment is capped. Used by navBar.jsp to hide "Add Participant" link in the menu
+
+            //If user logged in using an expired state cookie, bring them to the login redirect.
+            if (RequestContextHolder.getRequestAttributes().getAttribute("StateCookieInvalid", RequestAttributes.SCOPE_SESSION) != null) {
+                if ((boolean) RequestContextHolder.getRequestAttributes().getAttribute("StateCookieInvalid", RequestAttributes.SCOPE_SESSION)) {
```

Comment:
Instead of checking for null first and then for the actual value. You can store this value in a variable like "isStateCookieInvalid" and then use BooleanUtils.isTrue(isStateCookieInvalid). BooleanUtils will handle the null check as well.